src/Data/List/Lazy.purs

```diff
@@ -742,12 +742,12 @@ transpose xs =
 --------------------------------------------------------------------------------
 
 -- | Perform a fold using a monadic step function.
-foldM :: forall m a b. Monad m => (a -> b -> m a) -> a -> List b -> m a
-foldM f a xs =
+foldM :: forall m a b. Monad m => (b -> a -> m b) -> b -> List a -> m b
+foldM f b0 xs =
     case uncons xs of
-         Nothing -> pure a
-         Just { head: b, tail: bs } ->
-                       f a b >>= \a' -> foldM f a' bs
+         Nothing -> pure b0
+         Just { head: a, tail: as } ->
+                       f b0 a >>= \b' -> foldM f b' as
```

Comment:
Just curious, but why did you use `b0` here?

Reply:
Yea, can we just do

```purs
foldM f b xs = ...
  Nothing -> pure b
  Just { head: a, tail: as } ->
    f b a >>= \b' -> foldM f b' as
```
```